### DISCUSSION TITLE
deploying a solidstart app results in a 404

### DISCUSSION THREAD

**Discussion by stefan-karger** [vercel/community#2373] [Help] [answered]
### Summary

I recently switched from nextjs to solidjs for my personal project.
But even after deleting and completely recreating the vercel project I get a 404:

![grafik](https://user-images.githubusercontent.com/8871906/237055219-5378ed3b-4556-4e6a-ab6e-5d635600c88e.png)

these are my deploy settings inside vercel:

![grafik](https://user-images.githubusercontent.com/8871906/237055579-b7ef6744-f131-4b44-a0fb-6b8279b8b951.png)

deployments work without any error:

![grafik](https://user-images.githubusercontent.com/8871906/237055815-dae9167a-1371-4612-8784-e38651630f8a.png)

but it works completely fine locally both as dev and build:

![grafik](https://user-images.githubusercontent.com/8871906/237058080-11e1939d-38ca-4f20-b4d1-c1441bf3a7a2.png)

if you need more information please let me know

### Additional information

github: https://github.com/sek-consulting/social-crm
vercel url: https://social-crm.vercel.app/

### Example

https://github.com/sek-consulting/social-crm

**Comment by stefan-karger** [comment]
same error is easily reproduced by simply trying to deploy a bare solid-start app...
works totally fine locally but results in a 404 with the following changes to the settings:

![grafik](https://github.com/vercel/community/assets/8871906/73184116-6b6c-44bb-8e03-8bb47f14ff77)

github: https://github.com/sek-consulting/solid-vercel-test
vercel url: https://solid-vercel-test-dun.vercel.app/

![grafik](https://github.com/vercel/community/assets/8871906/3428791d-2475-4209-bf2a-cb15828b1f35)

with directory listing turned on you can see the following:

![grafik](https://github.com/vercel/community/assets/8871906/a36804d7-1970-4b9c-a181-604893c71a0f)

**Comment by amyegan** [comment, marked as answer]
I was able to deploy the [Vercel SolidStart example](https://github.com/vercel/vercel/tree/main/examples/solidstart) without `404` or other errors. Then, I tried with a different app by following the "Using a starter" instructions from https://start.solidjs.com/getting-started/project-setup. With that app, I did get a build error with the default settings and a `404` when I changed the Output Directory to `dist`. 

Comparing the two, I noticed that the first example uses the [Vercel Adapter](https://github.com/solidjs/solid-start/tree/main/packages/start-vercel) for SolidStart. I tried adding [`solid-start-vercel`](https://github.com/solidjs/solid-start/tree/main/packages/start-vercel) to the "Using a starter" example, and that got it working. 

Does your app work when you use the adapter?

**Reply by stefan-karger** [reply]
Thank you so much for that hint!
Now it works just fine :)
